scripts/pylib/twister/twisterlib/twister_main.py

```diff
@@ -201,12 +205,12 @@ def main(options):
 
     coverage_completed = True
     if options.coverage:
-        if not options.build_only:
+        if not options.build_only or not options.report_summary:
```

Comment:
why is this change ?

Reply:
because it may not work properly with coverage

Reply:
> because it may not work properly with coverage

what is wrong with the new option and coverage ?

Reply:
> > because it may not work properly with coverage
> 
> what is wrong with the new option and coverage ?

Never mind, I deleted this. The main function for report_summary returns 0, so coverage never started.